src/modules/cpu/kernel/normalize_generic.hpp

```diff
@@ -0,0 +1,513 @@
+    Rpp32u bufferLength = dims[1];
+    Rpp32u alignedLength = (bufferLength / 8) * 8;
+    Rpp32u numRows = dims[0];
+
+    __m256 pShift = _mm256_set1_ps(shift);
+    for(Rpp32u i = 0; i < numRows; i++)
+    {
+        Rpp32f *srcPtrTempRow = srcPtrTemp + i * srcDescPtr->strides[1];
+        Rpp32f *dstPtrTempRow = dstPtrTemp + i * dstDescPtr->strides[1];
+
+        // set mean and stddev
+        Rpp32f mean = meanPtr[i];
+        Rpp32f invStdDev = invStdDevPtr[i];
+        __m256 pMean, pInvStdDev;
+        pMean = _mm256_set1_ps(mean);
+        pInvStdDev = _mm256_set1_ps(invStdDev);
+
+        Rpp32u vectorLoopCount = 0;
+        for(; vectorLoopCount < alignedLength ; vectorLoopCount += 8)
+        {
+            __m256 pSrc = _mm256_loadu_ps(srcPtrTempRow);
+            __m256 pDst = _mm256_add_ps(_mm256_mul_ps(_mm256_sub_ps(pSrc, pMean), pInvStdDev), pShift);
+            _mm256_storeu_ps(dstPtrTempRow, pDst);
+            srcPtrTempRow += 8;
+            dstPtrTempRow += 8;
+        }
+        for(; vectorLoopCount < dims[1] ; vectorLoopCount += 8)
+             *dstPtrTempRow++ = (*srcPtrTempRow++ - mean) * invStdDev + shift;
+    }
+}
+
+RppStatus normalize_generic_f32_f32_host_tensor(Rpp32f *srcPtr,
+                                                RpptGenericDescPtr srcGenericDescPtr,
+                                                Rpp32f *dstPtr,
+                                                RpptGenericDescPtr dstGenericDescPtr,
+                                                Rpp32u axis_mask,
+                                                Rpp32f *meanTensor,
+                                                Rpp32f *stdDevTensor,
+                                                Rpp32u computeMean,
+                                                Rpp32u computeStddev,
+                                                Rpp32f scale,
+                                                Rpp32f shift,
+                                                Rpp32u *roiTensor,
+                                                RppLayoutParams layoutParams,
+                                                rpp::Handle& handle)
+{
+	Rpp32u numThreads = handle.GetNumThreads();
+    Rpp32u nDim = srcGenericDescPtr->numDims;
+    Rpp32u batchSize = dstGenericDescPtr->dims[0];
+
+    omp_set_dynamic(0);
+#pragma omp parallel for num_threads(numThreads)
+	for(int batchCount = 0; batchCount < batchSize; batchCount++)
+	{
+        Rpp32f *srcPtrTemp, *dstPtrTemp;
+        srcPtrTemp = srcPtr + batchCount * srcGenericDescPtr->strides[0];
+        dstPtrTemp = dstPtr + batchCount * dstGenericDescPtr->strides[0];
+
+        // Set all values in dst buffer to 0.0
+        for(int cnt = 0; cnt < dstGenericDescPtr->strides[0]; cnt++)
+            dstPtrTemp[cnt] = 0.0f;
+
+        Rpp32u *roi = roiTensor + batchCount * nDim * 2;
+        Rpp32u *begin = roi;
+        Rpp32u *length = &roi[nDim];
+
+        Rpp32u paramStride[nDim];
```

Comment:
Please allocate this dynamically using malloc

Reply:
Sure